src/ucp/core/ucp_ep.c

```diff
@@ -318,20 +318,27 @@ ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, unsigned ep_init_flags,
         goto out;
     }
 
-    if ((context->config.ext.proto_indirect_id == UCS_CONFIG_ON) ||
-        ((context->config.ext.proto_indirect_id == UCS_CONFIG_AUTO) &&
-         (ep_init_flags & UCP_EP_INIT_ERR_MODE_PEER_FAILURE))) {
-        ucp_ep_update_flags(ep, UCP_EP_FLAG_INDIRECT_ID, 0);
-    }
+    if (!(ep_init_flags & UCP_EP_INIT_FLAG_CM_TMP)) {
+        if ((context->config.ext.proto_indirect_id == UCS_CONFIG_ON) ||
+            ((context->config.ext.proto_indirect_id == UCS_CONFIG_AUTO) &&
+             (ep_init_flags & UCP_EP_INIT_ERR_MODE_PEER_FAILURE))) {
+            ucp_ep_update_flags(ep, UCP_EP_FLAG_INDIRECT_ID, 0);
+        }
 
-    status = ucs_ptr_map_put(&worker->ptr_map, ep,
-                             !!(ep->flags & UCP_EP_FLAG_INDIRECT_ID),
-                             &ucp_ep_ext_control(ep)->local_ep_id);
-    if (status != UCS_OK) {
-        goto err_destroy_ep_base;
+        status = ucs_ptr_map_put(&worker->ptr_map, ep,
+                                 !!(ep->flags & UCP_EP_FLAG_INDIRECT_ID),
+                                 &ucp_ep_ext_control(ep)->local_ep_id);
+        if (status != UCS_OK) {
+            goto err_destroy_ep_base;
+        }
     }
 
-    ucs_list_add_tail(&worker->all_eps, &ucp_ep_ext_gen(ep)->ep_list);
+    if (ep_init_flags & (UCP_EP_INIT_FLAG_REPLY | UCP_EP_INIT_FLAG_CM_TMP)) {
```

Comment:
- UCP_EP_INIT_FLAG_INTERNAL. no need id for WIREUP_REMOVED endpoint
- memtype endpoints on intenal_eps list